### DISCUSSION TITLE
Iomega Alpha ( 10 / 20 MB Bernoulli ) Disks, where to even start?

### DISCUSSION THREAD

**Discussion by Bob-O-Rama** [davidgiven/fluxengine#776] [General]
The drives use a flexible disk spinning at 1500 RPM which forms a fluid bearing over a "bernoulli plate" making an essentially  rigid disk platter when spun up.   The data clock rate for these drives in about 13Mbps raw, exactly 50% higher than the bit rate of stored data.   On the scope I am seeing a vaguely sinusodial-ish signal at 1/4th of that 13Mhz frequency.  This is too fast for any of the flux reading devices at my disposal.  The  Iomega documentation says it uses a NRZ and RLLC for encoding.   I just don't know what that "should" look like.  

Obvious if there is a better place to discuss this splunking effort, let me know.   But ideally I'd like to be able to do the analysis with this or a similar project.   I have become very expert in fixing / maintaining / recovering data from these devices - and can likely also write the code to analyze and decode the signals - so I have the tools and expertise - but I'm not smart enough to see how anything is being stored in the data fields.     The synchro, servo, and what I think are sector marks - they use a somewhat slower symbol rate and look very much like the series of asymptotes you see with floppy disks.   But the rest?   Its just mysterious.

**Comment by davidgiven** [comment]
That's a completely different world than the stuff I'm working with!

I'm guessing that you're trying to read these directly from the analogue circuitry, rather than going through the Iomega drive controller? Which, if it was anything like the Iomega devices I've seen, was based on SCSI. Are you at least downstream of whatever head amplifier the system is using? Assuming it's based on the same magnetic flux reversals that floppy disks are, that should do a lot of the heavy lifting for you.

NRZ suggests that the actual polarity of the signal is relevant, unlike floppy disks where only the timing between reversals is relevant. But everything I know about it came from Wikipedia in the last five minutes. It does suggest it's not self-clocking. Do you know if there's an additional signal providing an external clock?

**Reply by Bob-O-Rama** [reply]
I've made some progress since I posted.   The controller, when writing, sends a pulse train to a well documented disk R/W preamp chip.   That chip has a flipflop withich alternates which half of the R/W head is energized while writing.   So write current is on throughout, and the chip uses a series of pulses from the controller to cause a flux reversal. 

 The pulse code is a 2 into 3 RLL(1,8) encoding scheme which I now have reverse engineered.   At this point I can encode / decode that _digital pulse train_ the controller uses to trigger flux reversals back and forth to the actual encoded user data.    So this has informed me what the flux reversal patterns should be when read - and I see stuff that looks plausible on a sampling oscilloscope.    

Its not self clocking, there is 160 - 230 pulse square wave sequence ( ...10101010101... in encoded bits, and so flux reversals ) which is used to sync up a VFO.   That VFO manages the detection windows for peaks and valleys.   My own _write decoding_ logic uses the VFO field to identify fata fields, and it works flawlessly, and I can reconstruct the data being written to the disk.   So I have the digital encoding figured out.

On the read side of things I am capturing the 100mV P-P  analog signals ( a differential signal ) from the preamplifier to the analog section of the controller.   There are easily discernible peaks on top of a really noisy signal.   So I'm trying my  best to do peak detection without any knowledge of how to do that, LOL.   I'm taking the 250Ms/sec AD capture from my scope and trying to turn that back into the peaks, and then reconstruct the pulse train - which as mentioned I can already decode.   Since I know the actual data, and the pulse train that write it, it "should" be easy.

Now that I've "cracked the code" on the digital side, my biggest issue is getting a rigorous peak detection method that would accepts the oscilloscopes AD capture.   For flux reads  I am assuming that is not as simple as following the voltage measured?   Is there some magick to identifying the "center" of a peak?    What is the proper way to do this?   I do know flux reversals should occur on the trailing edge of pulses synced to to a 13.608Mhz clock - so maybe this is the game, to try a windowing mechanism and just look for peak / trough only when there "should" be one.

**Reply by Bob-O-Rama** [reply]
But yeah, to be clear, my hope was to use / enhance the project to treat this like a "big" floppy - and I have already worked out a lot of the electronic "stuff" to take the mechanics and drive them without a controller.   The ultimate goal is to do low level formatting of them - while the high level sector IDs and data fields are overwritten when a SCSI format command is issued or when a sector is overwritten - the low level formatting is not and the flux is so weak on many cartridges that the head positioning servo cannot accurately do its job - even if the data fields are readable.   What I hope to do is use a stepper to control the absolute head position, and blast the disk with new servo marks for each track.   Then a high level format should re-fresh the rest.

**Comment by davidgiven** [comment]
(also, converted to a discussion thread rather than an issue, but forgot to say so first. Hopefully this reply will ping you with the right URL)
